Server/Components/LegacyNetwork/legacy_network_impl.cpp

```diff
@@ -451,7 +452,9 @@ void RakNetLegacyNetwork::OnPlayerConnect(RakNet::RPCParameters* rpcParams, void
 
 				network->networkEventDispatcher.dispatch(&NetworkEventHandler::onPeerConnect, *newPeer);
 
-				network->playerRemoteSystem[newPeer->getID()] = remoteSystem;
+				auto poolID = newPeer->getID();
```

Comment:
This doesn't make sense to me - if a new peer was just created successfully (non null) then their ID should be valid